Sources/NIOTransportServices/NIOTSEventLoop.swift

```diff
@@ -262,4 +262,21 @@ extension NIOTSEventLoop {
         assert(oldChannel != nil)
     }
 }
+
+// MARK: SerialExecutor conformance
+#if compiler(>=5.9)
+@available(macOS 14.0, *)
+extension NIOTSEventLoop: NIOSerialEventLoopExecutor {
+    func enqueue(_ job: consuming ExecutorJob) {
+        let unownedJob = UnownedJob(job)
+        if let executorQueue = self.taskQueue as? _DispatchSerialExecutorQueue {
```

Comment:
This is currently just available as underscored API and cannot be relied on. This PR is more for demonstrating that this can be fast pathed as well.